src/shared/sam/cli/samCliInit.ts

```diff
@@ -20,25 +18,12 @@ export async function runSamCliInit(
     initArguments: SamCliInitArgs,
     invoker: SamCliProcessInvoker
 ): Promise<void> {
-    const { exitCode, error, stderr, stdout }: ChildProcessResult = await invoker.invoke(
+    const childProcessResult = await invoker.invoke(
         { cwd: initArguments.location },
         'init',
         '--name', initArguments.name,
         '--runtime', initArguments.runtime
     )
 
-    if (exitCode === 0) {
-        return
-    }
-
-    console.error('SAM CLI error')
-    console.error(`Exit code: ${exitCode}`)
-    console.error(`Error: ${error}`)
-    console.error(`stderr: ${stderr}`)
-    console.error(`stdout: ${stdout}`)
-
-    const logger: Logger = getLogger()
-    const err = new Error(`sam init encountered an error: ${error && error.message || stderr || stdout}`)
-    logger.error(err)
-    throw err
+    logAndThrowIfUnexpectedExitCode(childProcessResult, 0)
```

Comment:
I feel like you should leave out the `, 0`. It feels like that should be understood.

At what points would we expect something other than 0?

Reply:
Best to be explicit and make no assumptions about what default `logAndThrowIfUnexpectedExitCode` contains.

Reply:
Thoughts on something like:
```
export function logAndThrowIfError(childProcessResult) {
    logAndThrowIfUnexpectedExitCode(childProcessResult, 0)
}
```

It just doesn't feel like there's much reason to watch for anything aside from 0.